docs/FormActions.md

```diff
@@ -0,0 +1,81 @@
+# Form Actions
+
+## Description
+
+In order to create usable forms, the components need to be able to integrate with standard Canvas App controls so that standard form actions can be executed. 
+
+The components can be hooked up to Canvas App controls via exposed custom input properties. We will focus on the **submit** and **reset** actions of our components.
+
+The behaviour and instructions described here are the same for all of the components.
+
+## Submit
+
+### Description
+
+Every component has a boolean input **Submit enabled** which dictates whether the component uses live validation (boolean=false) or on submit validation (boolean=true). 
+
+Live validation means the validation occurs as soon as the input is clicked out of. This means the errors show as soon as the **HasError** output boolean switches from false to true. For an error to show for empty field validation, a value must be entered, clicked out of, clicked back into, removed and then clicked out of again. This prevents errors showing as soon as fields are clicked into. 
```

Comment:
`a value must be entered, clicked out of, clicked back into, removed and then clicked out of again...` -> `a value must have been entered and then removed, to prevent errors showing before being interacted with` ?

Reply:
this isn't strictly true, the clicking in and out is required for the error to show. I have amended though since its much easier to understand